xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroParameterDescriptor.java

```diff
@@ -163,4 +167,145 @@ public boolean isMandatory()
     {
         return mandatory;
     }
+
+    @Override
+    public boolean isAdvanced()
+    {
+        return advanced;
+    }
+
+    /**
+     * Allows to set whether the parameter is advanced or not. See {@link #isAdvanced()}.
+     * @param advanced {@code true} if the parameter is advanced.
+     * @return the current instance (builder pattern)
+     * @since 17.3.0RC1
+     * @since 16.10.6
+     */
+    @Unstable
+    public WikiMacroParameterDescriptor setAdvanced(boolean advanced)
```

Comment:
Not a huge fan of suddenly making `WikiMacroParameterDescriptor` mutable. Note that anyone with script right can call those setters right now, so better be safe.